using-mobile-extensions/adobe-experience-platform-messaging/README.md

```diff
@@ -0,0 +1,211 @@
+5. Drag and drop the **Apple Push Notification Authentication Key** for your Apple developer account. This key can be acquired from the **Certificates, Identifiers & Profiles** page. 
+6. Provide the **Key ID**. This is a 10 character string assigned during the creation of p8 auth key. It can be found under **Keys** tab in **Certificates, Identifiers & Profiles** page.
+7. Provide the **Team ID**. This is a string value which can be found under the Membership tab.
+
+![AEP Android App Configuration](../../.gitbook/assets/add-app-config-ios.png)
+
+For more information follow this [link](https://help.apple.com/developer-account/#/deva05921840)
+
+{% endtab %}
+{% endtabs %}
+
+## Configure the application to allow push notification 
+{% tabs %}
+{% tab title="Android" %}
+To add Firebase Messaging Service to your app please follow the [link](https://firebase.google.com/docs/cloud-messaging/android/client)
+
+{% endtab %}
+
+{% tab title="iOS" %}
+To add APNs to your app please follow the [link](https://developer.apple.com/documentation/usernotifications/registering_your_app_with_apns)
+
+{% endtab %}
+{% endtabs %}
+
+## Add the AEP Messaging extension to your application
+
+### Download and import the Messaging extension
+
+{% tabs %}
+{% tab title="Android" %}
+
+### Java
+
+1. Add the Mobile Core, Edge and Messaging extensions to your project using the app's Gradle file.
+
+   ```java
+   implementation 'com.adobe.marketing.mobile:core:1.+'
+   implementation 'com.adobe.marketing.mobile:edge:1.+'
+   implementation 'com.adobe.marketing.mobile:edgeidentity:1.+'
+   implementation 'com.adobe.marketing.mobile:messaging:1.+'
+   ```
+
+2. Import the Mobile Core Edge and Messaging extensions in your application class.
+
+   ```java
+    import com.adobe.marketing.mobile.*;
+   ```
+
+
+{% endtab %}
+
+{% tab title="iOS" %}
+
+1. Add the Mobile Core, Edge and Messaging extensions to your project using Cocoapods. Add following pods in your `Podfile`:
+
+   ```swift
+   use_frameworks!
+   target 'YourTargetApp' do
+       pod 'AEPCore'
+       pod 'AEPEdge'
+       pod 'AEPEdgeIdentity'
+       pod 'AEPMessaging'
+   end
+   ```
+
+2. Import the Mobile Core, Edge and Messaging libraries:
+
+### Swift
+
+```swift
+// AppDelegate.swift
+import AEPCore
+import AEPEdge
+import AEPEdgeIdentity
+import AEPMessaging
+```
+
+### Objective-C
+
+```text
+// AppDelegate.h
+@import AEPCore;
+@import AEPEdge;
+@import AEPEdgeIdentity;
+@import AEPMessaging;
+```
+
+{% endtab %}
+{% endtabs %}
+
+### Register Messaging extensions with Mobile Core
+
+{% tabs %}
+{% tab title="Android" %}
+
+### Java
+
+```java
+public class MobileApp extends Application {
+    @Override
+    public void onCreate() {
+      super.onCreate();
+      MobileCore.setApplication(this);
+      MobileCore.configureWithAppID("yourLaunchEnvironmentID");
+      try {
+        Edge.registerExtension();
+        Identity.registerExtension();
```

Comment:
does this properly register `EdgeIdentity`?  will there be conflict (or ambiguity) with the direct Identity since we are using a package import here?

Reply:
hmmm u r right it will so i ill add another import for edgeidentity in the doc above

Reply:
doesn't `import com.adobe.marketing.mobile.*` also include the other `Identity` class?  i think it will still complain about ambiguity